// @builder.io/sdk-react/dist/components/block/components/block-wrapper.js
"use client";
import * as React from "react";
import { getBlockActions } from "../../../functions/get-block-actions.js";
import { getBlockProperties } from "../../../functions/get-block-properties.js";
function BlockWrapper(props) {
    return (React.createElement(React.Fragment, null,
        " ",
        props.hasChildren ? (React.createElement(React.Fragment, null,
            React.createElement(props.Wrapper, { ...getBlockProperties({
                    block: props.block,
                    context: props.context,
                }), ...getBlockActions({
                    block: props.block,
                    rootState: props.context.rootState,
                    rootSetState: props.context.rootSetState,
                    localState: props.context.localState,
                    context: props.context.context,
                    stripPrefix: true,
                }) },
                " ",
                props.children,
                " "))) : (React.createElement(React.Fragment, null,
            React.createElement(props.Wrapper, { ...getBlockProperties({
                    block: props.block,
                    context: props.context,
                }), ...getBlockActions({
                    block: props.block,
                    rootState: props.context.rootState,
                    rootSetState: props.context.rootSetState,
                    localState: props.context.localState,
                    context: props.context.context,
                    stripPrefix: true,
                }) }))),
        " "));
}
export default BlockWrapper;
