// @builder.io/sdk-react/dist/components/blocks/blocks-wrapper.js
"use client";
import * as React from "react";
import { isEditing } from "../../functions/is-editing.js";
function BlocksWrapper(props) {
    function className() {
        return "builder-blocks" + (!props.blocks?.length ? " no-blocks" : "");
    }
    function onClick() {
        if (isEditing() && !props.blocks?.length) {
            window.parent?.postMessage({
                type: "builder.clickEmptyBlocks",
                data: {
                    parentElementId: props.parent,
                    dataPath: props.path,
                },
            }, "*");
        }
    }
    function onMouseEnter() {
        if (isEditing() && !props.blocks?.length) {
            window.parent?.postMessage({
                type: "builder.hoverEmptyBlocks",
                data: {
                    parentElementId: props.parent,
                    dataPath: props.path,
                },
            }, "*");
        }
    }
    return (React.createElement(React.Fragment, null,
        React.createElement("div", { className: className() + " div-5821bd52", "builder-path": props.path, "builder-parent-id": props.parent, ...{}, style: props.styleProp, onClick: (event) => onClick(), onMouseEnter: (event) => onMouseEnter(), onKeyPress: (event) => onClick() }, props.children),
        React.createElement("style", null, `.div-5821bd52 {
  display: flex;
  flex-direction: column;
  align-items: stretch;
}`)));
}
export default BlocksWrapper;
